src/UI/UIView.cpp

```diff
@@ -787,37 +860,73 @@ glm::vec3 UIView::get_clip_max()
 
 void UIView::set_chracter(char c)
 {
-  if (m_modal_trnfct_open)
+  if (m_trn_fct_name_open)
   {
     m_copy_trnfnct_name += c;
   }
-  if (m_save_trnfct_open)
+  if (m_save_trnfct_open || m_save_session_dialog_open)
   {
     m_save_file_name += c;
   }
 }
 
 void UIView::remove_character()
 {
-  if (m_modal_trnfct_open)
+  if (m_trn_fct_name_open )
   {
     m_copy_trnfnct_name.pop_back();
   }
   
-  if (m_save_trnfct_open)
+  if (m_save_trnfct_open || m_save_session_dialog_open)
   {
     m_save_file_name.pop_back();
   }
 }
 
+void UIView::open_save_modal_dialog(std::string& id, bool& window_state, std::function<void()> save_function)
+{
+  ImGui::OpenPopup(id.c_str());
+  ImGui::SetNextWindowSize(ImVec2(350, 200), ImGuiCond_FirstUseEver);
+  if (ImGui::BeginPopupModal(id.c_str(), &window_state))
+  {
+    char* writable = new char[m_save_file_name.size() + 1];
+    std::copy(m_save_file_name.begin(), m_save_file_name.end(), writable);
+    writable[m_save_file_name.size()] = '\0';
+    ImGui::Text(m_dir_to_save.c_str());
+    ImGui::SameLine();
+    if (ImGui::Button("..."))
+    {
+      m_file_dialog_save_dir = true;
+    }
+
+    ImGui::InputText("##text2", writable, IM_ARRAYSIZE(writable));
```

Comment:
what is `##text2`?

Reply:
It's a random ID the UI for that component. I took it from an old example. I rename it to something more meaningful.